client/src/components/Minimap/index.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { MinimapMenuItem } from './MinimapItem';
+
+import { toggleCollapsiblePanel } from '../../includes/panels';
+import { debounce } from '../../utils/debounce';
+
+import Minimap from './Minimap';
+
+/**
+ * Generate a minimap link’s data, based on the panel’s elements.
+ */
+const createMinimapLink = (
+  anchor: HTMLAnchorElement,
+): MinimapMenuItem | null => {
+  const panel = anchor.closest<HTMLElement>('[data-panel]');
+  const headingId = panel?.getAttribute('aria-labelledby');
+  const heading = panel?.querySelector<HTMLHeadingElement>(`#${headingId}`);
+  const toggle = panel?.querySelector<HTMLButtonElement>('[data-panel-toggle]');
+  // Special case for InlinePanel, where deleted items are kept until the form is saved.
+  const inlinePanelDeleted = anchor.closest(
+    '[data-inline-panel-child].deleted',
+  );
+  if (!panel || !heading || !toggle || inlinePanelDeleted) {
+    return null;
+  }
+
+  const headingText = heading.querySelector('[data-panel-heading-text]');
+  // If the heading’s most correct text content is unavailable (StreamField block collapsed when empty),
+  // fall back to the full heading text.
+  const label =
+    headingText?.textContent ||
+    heading.textContent?.replace(/\s+\*\s+$/g, '').trim();
+  const required = panel.querySelector('[data-panel-required]') !== null;
+  const useElt = toggle.querySelector<SVGUseElement>('use');
+  const icon = useElt?.getAttribute('href')?.replace('#icon-', '') || '';
+  const ariaLevel = heading.getAttribute('aria-level');
+  const headingLevel = `h${ariaLevel || heading.tagName[1] || 2}`;
+  const errorCount = [].slice
+    .call(panel.querySelectorAll('.error-message'))
+    .filter((err) => err.closest('[data-panel]') === panel).length;
```

Comment:
This would warrant a comment – it’s important we only count a panel’s errors if the errors aren’t inside any other nested panel. So -- report the errors on the innermost panel.